unliftio/src/UnliftIO/IO/File.hs

```diff
@@ -119,8 +123,30 @@ writeBinaryFileAtomic fp bytes =
 withBinaryFileDurable = Posix.withBinaryFileDurable
 withBinaryFileDurableAtomic = Posix.withBinaryFileDurableAtomic
 withBinaryFileAtomic = Posix.withBinaryFileAtomic
+
+renameFileDurable = Posix.renameFileDurable
 #endif
 
+-- | When a file is renamed, it is necessary to execute @fsync()@ on the
+-- directory that contains the file now and afterwards on the directory where
+-- the file was before, so that the rename is durable.
+--
+-- Remark: This is also atomic if both locations of the file are on the same
+-- filesystem. However, it could happen that the operation leads to data loss,
+-- if a crash happens after the rename and before the first fsync finishes. This
+-- is because on an async filesystem the write of the old directory might
+-- already written to disk and the change on the new directory is not. It the
+-- function call returns, the change is durable. Nevertheless, this will not
+-- happen on filesystems using journaling, that is, allmost all modern filesystems.
```

Comment:
Is it not a bit confusing to say "This is **also** atomic"? Because it should be **only** atomic on the same filesystem -- `rename()` does not work at all across different file systems (on Linux), where we get from `System.Directory.renameFile`:

```
Prelude System.Directory> renameFile "testfile" "mnt/testfile"
*** Exception: testfile: renameFile:renamePath:rename: unsupported operation (Invalid cross-device link)
```